### PR TITLE
Rem 386

### PR DESCRIPTION
**Pull request by FatimaSalami**
in this task
filters were migrated from asp to react into apiMapping (combo, lookups)
changes to reportParamBrowser (in report param bug of date also) and Toolbar were made to solve bugs 
filters were added in different screens
change in customLookup was made to be able to display multiple fields in the displayField

a new way of appending params to 3 filters (doc type and dimensions)  was made in reportParams also

### CODE REVIEW

File: src/components/Inputs/CustomLookup.js [hunk 1]
```diff
@@ -73,10 +73,22 @@ const CustomLookup = ({
...
+
+              return text || firstValue
+            }
+
+            return typeof option === 'object' ? `${option[valueField] ? option[valueField] : ''}` : option
+          }}
+          
+          // isOptionEqualToValue={(option, value) =>
+          //   value && !Array.isArray(valueField)  ? option[valueField] === value[valueField] : ''
+          // }
```
Comment (RebeccaFrem02) [inline src/components/Inputs/CustomLookup.js:91]: are those comments neccesary?

File: src/components/Inputs/CustomLookup.js [hunk 2]
```diff
@@ -73,10 +73,22 @@ const CustomLookup = ({
...
-          getOptionLabel={option =>
-            typeof option === 'object' ? `${option[valueField] ? option[valueField] : ''}` : option
-          }
-          isOptionEqualToValue={(option, value) => (value ? option[valueField] === value[valueField] : '')}
+          getOptionLabel={option => {
+            if (typeof valueField == 'object') {
+              const text = valueField
+                .map(header => option[header] && option[header]?.toString())
+                ?.filter(item => item)
+                ?.join(' ')
```
Comment (RebeccaFrem02) [inline src/components/Inputs/CustomLookup.js:81]: remove empty string

File: src/components/Shared/apiMappings.js [hunk 3]
```diff
@@ -41,60 +49,71 @@ export const apiMappings = {
...
-    valueField: 'reference',
+    firstField: 'reference',
     valueOnSelection: 'recordId',
-    displayField: 'name',
+    secondField: 'name',
     displayFieldWidth: 2,
     firstFieldWidth: '40%'
   },
   [ResourceIds.Category]: {
     //Item Category
```
Comment (RebeccaFrem02) [inline src/components/Shared/apiMappings.js:59]: are the comments here neccesary?

File: src/components/Shared/apiMappings.js [hunk 4]
```diff
@@ -41,60 +49,71 @@ export const apiMappings = {
...
     valueOnSelection: 'recordId',
-    displayField: 'name',
     displayFieldWidth: 2,
     firstFieldWidth: '40%'
   },
   [ResourceIds.DocumentTypes]: {
     type: COMBOBOX,
     endpoint: SystemRepository.DocumentType.qry,
-    parameters: '_filter=',
+    parameters: '_startAt=0&_pageSize=2000', //_dgId appended
```
Comment (RebeccaFrem02) [inline src/components/Shared/apiMappings.js:116]: remove //_dgId appended

File: src/components/Shared/apiMappings.js [hunk 5]
```diff
@@ -230,5 +248,414 @@ export const apiMappings = {
...
+    displayField: 'name'
+  },
+  [ResourceIds.Vehicle]: {
+    type: COMBOBOX,
+    endpoint: DeliveryRepository.Vehicle.qry,
+    valueField: 'recordId',
+    displayField: 'name'
+  },
+  [ResourceIds.SecurityGroupsFilter]: {
+    //true resourceId: SecurityGroup = 23102, security groups report asp broken!
```
Comment (RebeccaFrem02) [inline src/components/Shared/apiMappings.js:482]: remove

File: src/components/Shared/apiMappings.js [hunk 6]
```diff
@@ -230,5 +248,414 @@ export const apiMappings = {
...
+    endpoint: AccessControlRepository.NotificationLabel.qry,
+    valueField: 'recordId',
+    displayField: 'label'
+  },
+  [ResourceIds.Address]: {
+    type: LOOKUP,
+    endpoint: SystemRepository.Address.snapshot,
+    firstField: ['name', 'city', 'street1', 'email1', 'phone'],
+    
+    // secondField: 'city',
```
Comment (RebeccaFrem02) [inline src/components/Shared/apiMappings.js:622]: remove

File: src/components/Shared/apiMappings.js [hunk 7]
```diff
@@ -230,5 +248,414 @@ export const apiMappings = {
...
+    endpoint: GeneralLedgerRepository.CostCenterGroup.qry,
+    valueField: 'recordId',
+    displayField: ['reference', 'name'],
+    columnsInDropDown: [
+      { key: 'reference', value: 'Reference' },
+      { key: 'name', value: 'Name' }
+    ]
   }
+
+  //HR filters not migrated
```
Comment (RebeccaFrem02) [inline src/components/Shared/apiMappings.js:660]: remove all comments

File: src/pages/journal-vouchers/index.js [hunk 8]
```diff
@@ -122,17 +124,36 @@ const JournalVoucher = () => {
     toast.success('Record Deleted Successfully')
```
Comment (RebeccaFrem02) [inline src/pages/journal-vouchers/index.js:124]: make sure that all the messages are platformLabels

File: src/pages/journal-vouchers/index.js [hunk 9]
```diff
@@ -122,17 +124,36 @@ const JournalVoucher = () => {
...
           maxAccess={access}
-          onSearch={search}
-          onSearchClear={clear}
-          labels={_labels}
-          inputSearch={true}
-        />{' '}
+          onApply={onApply}
+          onSearch={onSearch}
+          onClear={onClear}
+          reportName={'GLTR'}
```
Comment (RebeccaFrem02) [inline src/pages/journal-vouchers/index.js:155]: check if the reportName in all reports should be written manually or as label

File: src/pages/memos/[functionId]/index.js [hunk 10]
```diff
@@ -28,12 +29,14 @@ const Financial = () => {
   const { functionId } = router.query
 
   async function fetchGridData(options = {}) {
-    const { _startAt = 0, _pageSize = 50 } = options
+    const { _startAt = 0, _pageSize = 50, params } = options
     console.log('fetching')
```
Comment (RebeccaFrem02) [inline src/pages/memos/[functionId]/index.js:33]: remove console from all the pages

File: src/pages/memos/[functionId]/index.js [hunk 11]
```diff
@@ -46,25 +49,28 @@ const Financial = () => {
...
+    access,
+    invalidate
   } = useResourceQuery({
-    endpointId: FinancialRepository.FiMemo.qry,
+    endpointId: FinancialRepository.FiMemo.page,
     datasetId: ResourceIds.CreditNote,
 
     filter: {
-      endpointId: FinancialRepository.FiMemo.snapshot,
+      //endpointId: FinancialRepository.FiMemo.snapshot,
```
Comment (RebeccaFrem02) [inline src/pages/memos/[functionId]/index.js:59]: make sure that we don't need the endpoint here and remove it if not

File: src/repositories/FixedAssetsRepository.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    del: service + 'delGRP'
+  },
+  Location: {
+    qry: service + 'qryLOC',
+    get: service + 'getLOC',
+    set: service + 'setLOC',
+    del: service + 'delLOC'
+  },
+  Assets: {
+    //MDAssets
```
Comment (RebeccaFrem02) [inline src/repositories/FixedAssetsRepository.js:23]: remove the comment

File: src/resources/ResourceIds.js [hunk 13]
```diff
@@ -194,5 +194,30 @@ export const ResourceIds = {
   SystemFunctionIntegrations: 30105,
   PUPaymentTerms: 50106,
   PuVendors: 50201,
-  PUSettings: 50107
+  PUSettings: 50107,
+  Module: 20128, //filter
```
Comment (RebeccaFrem02) [inline src/resources/ResourceIds.js:198]: remove comment

File: src/resources/ResourceIds.js [hunk 14]
```diff
@@ -194,5 +194,30 @@ export const ResourceIds = {
...
+  Client: 51202,
+  SalesOrder: 51302,
+  Driver: 52201,
+  Vehicle: 52202,
+  SecurityGroupsFilter: 90101,
+  AssetClass: 43101,
+  AssetGroup: 43102,
+  Location: 43103,
+  MDAssets: 43200,
+  StockCounts: 44201, //stock count cycle
```
Comment (RebeccaFrem02) [inline src/resources/ResourceIds.js:216]: remove comment

File: src/resources/ResourceIds.js [hunk 15]
```diff
@@ -194,5 +194,30 @@ export const ResourceIds = {
...
+  Design: 42109,
+  GovernmentOrganizationFilter: 70106,
+  NotificationTransaction: 23301, // Notification center
```
Comment (RebeccaFrem02) [inline src/resources/ResourceIds.js:219]: remove comment

File: src/components/Inputs/CustomLookup.js (hunk 2 above)
Comment (FatimaSalami) [reply src/components/Inputs/CustomLookup.js:81]: we need the space between appearing fields

File: src/components/Shared/ReportParameterBrowser.js [hunk 16]
```diff
@@ -55,6 +61,13 @@ const GetLookup = ({ field, formik }) => {
         firstValue={formik.values.parameters?.[field.id]?.display}
         secondValue={formik.values.parameters?.[field.id]?.display2}
         onChange={(event, newValue) => {
+          const display = Array.isArray(apiDetails?.firstField)
+            ? apiDetails?.firstField
+                ?.map(header => newValue?.[header] && newValue?.[header]?.toString())
+                ?.filter(item => item)
+                ?.join(' ')
```
[the remark this answers is not in the document]
Comment (FatimaSalami) [reply src/components/Shared/ReportParameterBrowser.js:68]: we need the space between appearing fields

File: src/components/Shared/apiMappings.js (hunk 3 above)
Comment (FatimaSalami) [reply src/components/Shared/apiMappings.js:59]: yes prefer to have the lookups obvious

File: src/components/Shared/apiMappings.js (hunk 4 above)
Comment (FatimaSalami) [reply src/components/Shared/apiMappings.js:116]: prefer to keep them in the file

File: src/components/Shared/apiMappings.js (hunk 5 above)
Comment (FatimaSalami) [reply src/components/Shared/apiMappings.js:482]: prefer to keep them in the file

File: src/components/Shared/apiMappings.js (hunk 7 above)
Comment (FatimaSalami) [reply src/components/Shared/apiMappings.js:660]: prefer to keep them in the file

File: src/pages/journal-vouchers/index.js (hunk 8 above)
Comment (FatimaSalami) [reply src/pages/journal-vouchers/index.js:124]: will not refactor things in diff screens for now

File: src/pages/journal-vouchers/index.js (hunk 9 above)
Comment (FatimaSalami) [reply src/pages/journal-vouchers/index.js:155]: will check with nora
these are not used many times

File: src/resources/ResourceIds.js (hunk 13 above)
Comment (FatimaSalami) [reply src/resources/ResourceIds.js:198]: prefer to keep them in the file

File: src/resources/ResourceIds.js (hunk 14 above)
Comment (FatimaSalami) [reply src/resources/ResourceIds.js:216]: prefer to keep them in the file

File: src/resources/ResourceIds.js (hunk 15 above)
Comment (FatimaSalami) [reply src/resources/ResourceIds.js:219]: prefer to keep them in the file

File: src/components/Shared/ReportParameterBrowser.js [hunk 17]
```diff
@@ -4,6 +4,8 @@ import * as yup from 'yup'
 import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
```
Comment (Ghinwa-Arab) [inline src/components/Shared/ReportParameterBrowser.js:4]: remove yup import, not used

File: src/repositories/StockCountRepository.js [hunk 18]
```diff
@@ -0,0 +1,10 @@
+const service = 'SC.asmx/'
+
+export const StockCountRepository = {
```
Comment (Ghinwa-Arab) [inline src/repositories/StockCountRepository.js:3]: already we have stockCount repository, please remove this one and move endpoints to existing one

**Pull request merged**